docs/use_cases/social_media_retrieval.md

```diff
@@ -0,0 +1,679 @@
+
+```python
+class QdrantVectorDBRetriever:
+    def __init__(
+        self,
+        embedding_model: EmbeddingModelSingleton,
+        vector_db_client: QdrantClient,
+        cross_encoder_model: Optional[CrossEncoderModelSingleton] = None,
+        vector_db_collection: str = settings.VECTOR_DB_OUTPUT_COLLECTION_NAME,
+    ):
+        self._embedding_model = embedding_model
+        self._vector_db_client = vector_db_client
+        self._cross_encoder_model = cross_encoder_model
+        self._vector_db_collection = vector_db_collection
+
+    def search(
+        self, query: str, limit: int = 3, return_all: bool = False
+    ) -> Union[list[EmbeddedChunkedPost], dict[str, list]]:
+        ...
+
+    def embed_query(self, query: str) -> list[list[float]]:
+        ...
+
+    def rerank(self, query: str, posts: list[EmbeddedChunkedPost]) -> list[EmbeddedChunkedPost]:
+        ...
+
+    def render_as_html(self, post: EmbeddedChunkedPost) -> None:
+        ...    
+```
+
+### 5.1. Embed query
+
+We must embed the query precisely as we ingested the posts into the vector DB. Because the streaming pipeline is written in Python (due to Bytewax) and every preprocessing operation is modular, we can quickly replicate all steps necessary to embed the query.
+
+```python
+class QdrantVectorDBRetriever:
+
+    ...
+
+    def embed_query(self, query: str) -> list[list[float]]:
+        cleaned_query = CleanedPost.clean(query)
+        chunks = ChunkedPost.chunk(cleaned_query, self._embedding_model)
+        embdedded_queries = [
+            self._embedding_model(chunk, to_list=True) for chunk in chunks
+        ]
+
+        return embdedded_queries
+```
+
+Notice that we have multiple query chunks if the query is too large. That is not an issue. We can query Qdrant with each chunk and merge the results. It is even better, as we can search posts of interest in multiple areas in the embedded posts vector space.
+
+### 5.2. Plain retrieval
+
+After we preprocess the query, the retrieval step is straightforward. We must map every embedded query to a Qdrant `SearchRequest` object and call the `search_batch()` method on top of our LinkedIn posts collection.
+
+As we can end up with multiple queries (as we chunk the query if it's too long), when we merge all the retrieved posts, we must eliminate possible duplicates by adding all the items to a unique set based on their `chunk_id.`
+
+We will go into the `rerank` aspects of the method in just a second!
+
+Now, let's call our `QdrantVectorDBRetriever` class and see how it works:
+
+```python
+vector_db_retriever = QdrantVectorDBRetriever(embedding_model=EmbeddingModelSingleton(), vector_db_client=build_qdrant_client())
+
+retrieved_results = vector_db_retriever.search(query="Posts about Qdrant", limit=3, return_all=True)
+for post in retrieved_results["posts"]:
+    vector_db_retriever.render_as_html(post)
+```
+
+Here are the results ↓
+
+:::::tabs
```

Comment:
I am trying to show the images in different tabs using the `archbee` syntax. But it doesn't work directly on GitHub, so I want to be sure it will work on VectorHub.